src/base/global.h

```diff
@@ -48,6 +48,14 @@ constexpr typename std::add_const_t<T> asConst(T &&t) noexcept { return std::mov
 template <typename T>
 void asConst(const T &&) = delete;
 
+template <typename T, typename F, typename... Args>
+T modified(const T &t, F T:: *f, Args &&... args)
```

Comment:
IMO this isn't "modified" because you are making a copy of `t` and not modifying the original `t` directly.
You should take `t` as rvalue and modify it.

Reply:
Furthermore, I think this higher-order function isn't necessary. You could provide `Path & Path::removedExtension()` which would eliminate all potential implementation concerns.

Reply:
> IMO this isn't "modified" because you are making a copy of `t` and not modifying the original `t` directly. You should take `t` as rvalue and modify it, 

This is just a generalization of an existing interface, for example, in QString. It provides two methods for some actions, one of which modifies the current instance, and the other returns a modified copy, e.g. `chop()` and `chopped()`.  This template allows you to use second counterpart with the classes that provide only modifications (QString had only them until recently).

Reply:
> You could provide `Path & Path::removedExtension()` which would eliminate all potential implementation concerns.

No problem. But as I said before I don't want modify existing instance so it would be `Path Path::removedExtension()`.

Reply:
> But as I said before I don't want modify existing instance so it would be `Path Path::removedExtension()`.

OK.

Also another idea: `void removeExtension();` -> `Path &removeExtension();` so that one can chain the operations, for example: `Path("path/file.txt").removeExtension().toString();`

Reply:
> Also another idea: `void removeExtension();` -> `Path &removeExtension();` so that one can chain the operations, for example: `Path("path/file.txt").removeExtension().toString();`

IMO, it is generally good idea for modifying methods to return a reference (at least this would be consistent with the behavior of core operators like `+=`). But it is unlikely to be widely used in a "world" where in most cases we have a constant as input, and ubiquitous constancy is proposed as a fundamental principle.